File: app/src/main/java/net/opendasharchive/openarchive/media/PreviewMediaListActivity.java
package net.opendasharchive.openarchive.media;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import net.opendasharchive.openarchive.R;
import net.opendasharchive.openarchive.db.Media;
import net.opendasharchive.openarchive.fragments.MediaListFragment;
import net.opendasharchive.openarchive.publish.PublishService;
import net.opendasharchive.openarchive.util.Prefs;

import java.util.List;


public class PreviewMediaListActivity extends AppCompatActivity {

    MediaListFragment mFrag;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_preview_media);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setTitle(getString(R.string.title_activity_batch_media_review));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        mFrag = (MediaListFragment)getSupportFragmentManager().findFragmentById(R.id.fragUploadManager);

        showFirstTimeBatch();
    }


    @Override
    protected void onResume() {
        super.onResume();

        mFrag.refresh();

        mFrag.stopBatchMode();

    }

    @Override
    public void onPause() {
       
        super.onPause();
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {

        getMenuInflater().inflate(R.menu.menu_batch_review_media, menu);

        return true;
    }

    private void batchUpload ()
    {
        List<Media> listMedia = mFrag.getMediaList();

        for (Media media : listMedia)
        {
            media.status = Media.STATUS_QUEUED;
            media.save();
        }

        startService(new Intent(this, PublishService.class));
        finish();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {

            case R.id.menu_upload:
                batchUpload();
                return true;

            // Respond to the action bar's Up/Home button
            case android.R.id.home:
                //NavUtils.navigateUpFromSameTask(this);
                finish();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private void showFirstTimeBatch ()
    {

        if ( !Prefs.getBoolean("ft.batch")) {
            AlertDialog.Builder build = new AlertDialog.Builder(this, R.style.AlertDialogTheme)
                    .setTitle(R.string.popup_batch_title)
                    .setMessage(R.string.popup_batch_desc);


            build.create().show();

            Prefs.putBoolean("ft.batch",true);
        }
    }
}

File: app/src/main/java/net/opendasharchive/openarchive/features/media/PreviewMediaListActivity.kt
package net.opendasharchive.openarchive.features.media

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import net.opendasharchive.openarchive.R
import net.opendasharchive.openarchive.databinding.ActivityPreviewMediaBinding
import net.opendasharchive.openarchive.db.Media
import net.opendasharchive.openarchive.publish.PublishService
import net.opendasharchive.openarchive.util.Prefs

class PreviewMediaListActivity : AppCompatActivity() {

    private lateinit var mBinding: ActivityPreviewMediaBinding
    private var mFrag: MediaListFragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = ActivityPreviewMediaBinding.inflate(layoutInflater)
        setContentView(mBinding.root)
        initLayout()
        showFirstTimeBatch()
    }

    private fun initLayout() {
        setSupportActionBar(mBinding.toolbar)
        supportActionBar?.apply {
            title = resources.getString(R.string.title_activity_batch_media_review)
            setDisplayHomeAsUpEnabled(true)
        }
        mFrag = supportFragmentManager.findFragmentById(R.id.fragUploadManager) as? MediaListFragment
    }

    override fun onResume() {
        super.onResume()
        mFrag?.let {
            it.refresh()
            it.stopBatchMode()
        }
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_batch_review_media, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.menu_upload -> {
                batchUpload()
                return true
            }
            android.R.id.home -> {
                //NavUtils.navigateUpFromSameTask(this);
                finish()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    private fun batchUpload() {
        val listMedia = mFrag?.getMediaList() ?: listOf()
        for (media in listMedia) {
            media.status = Media.STATUS_QUEUED
            media.save()
        }
        startService(Intent(this, PublishService::class.java))
        finish()
    }

    private fun showFirstTimeBatch() {
        if (!Prefs.getBoolean("ft.batch")) {
            AlertDialog.Builder(this, R.style.AlertDialogTheme)
                    .setTitle(R.string.popup_batch_title)
                    .setMessage(R.string.popup_batch_desc).create().show()
            Prefs.putBoolean("ft.batch", true)
        }
    }

}